cli/src/lib/createStub.js

```diff
@@ -0,0 +1,178 @@
+/* @flow */
+
+import {signCode, verifySignedCode} from './codeSign';
+import {mkdirp} from './fileUtils';
+import {fs, path} from "./node";
+
+import {format} from 'util';
+import globAsync from 'glob';
+import colors from 'colors/safe';
+import resolveAsync from 'resolve';
+
+function glob(pattern: string, options: Object): Promise<Array<string>> {
+  return new Promise((resolve, reject) =>
+    globAsync(pattern, options, (err, files) => {
+      if (err) {
+        reject(err);
+      } else {
+        resolve(files);
+      }
+    })
+  );
+}
+
+function resolve(name: string, options: Object): Promise<string> {
+  return new Promise((resolve, reject) =>
+    resolveAsync(name, options, (err, result) => {
+      if (err) {
+        reject(err);
+      } else {
+        resolve(result);
+      }
+    })
+  );
+}
+
+const stubTemplate =
+`declare module '%s' {
+  declare module.exports: any;
+}`;
+
+function stubFor(moduleName: string): string {
+  return format(stubTemplate, moduleName);
+}
+
+async function writeStub(
+  projectRoot: string,
+  packageName: string,
+  version: string,
+  overwrite: boolean,
+  files: Array<string>,
+): Promise<string> {
+  let output = stubFor(packageName);
+  if (files.length > 0) {
+    output += format(
+      "\n\n%s\n",
+      files
+        .map(file => file.replace(/\.[^.]+$/, '')) // trim extensions
+        .map(file => stubFor(format("%s/%s", packageName, file)))
+        .join("\n\n"),
+    );
+  }
+  output += "\n"; // File should end with a newline
+  const filename = path.join(projectRoot, "flow-typed", packageName + ".js");
+  await mkdirp(path.dirname(filename));
+
+  if (!overwrite) {
+    const exists = await fs.exists(filename);
+    if (exists) {
+      const existingStub = await fs.readFile(filename);
+      if (!verifySignedCode(existingStub.toString())) {
+        throw new Error(
+          "Stub already exists and has been modified. " +
+          "Use --overwrite to overwrite",
+        );
+      }
+    }
+  }
+
+  await fs.writeFile(filename, signCode(output, version));
+  return filename;
+}
+
+/**
+ * createStub("/path/to/root", "foo") will create a file
+ * /path/to/root/flow-typed/foo.js that contains a stub for the module foo.
```

Comment:
My original though was to put something in `<PROJECT_ROOT>/flow-typed/stubs`. But I scrapped that idea based on the reasoning:

1. If the user never touches the stub, then it doesn't matter where it is.
2. If the user improves the stub, they'd probably want to move it to wherever they keep their custom libdefs. Which is probably not a stubs dir.

But I think your "if an official libdef shows up" argument is pretty convincing. I'll change dirs and add the version to the file name.